test/index.test.ts

```diff
@@ -286,11 +286,11 @@ describe('HipThrusTS', () => {
   });
 */
   describe('Hipthrusts functional only', () => {
-    // @todo: add test coverage for javascript side functionality too!
     // @todo: add test coverage for non-explicit-type-specification too!
```

Comment:
Good start...
- [ ] 1. instead of making entirely new it()s, why not just add the "act" and "assert" parts of these tests to the ends of each of the existing it()s so you don't have to repeat the setups and it's clearer and easier?
- [ ] 2. the outputs from the functions are being tested, but not the inputs...  One simple but not-super-clean way to do it is something like this for example:

```
        const aPassedIn = 'some string';
        const bPassedIn = 'another string';
        const left = {
￼          attachData(context: { a: string }) {
￼            expect(context.a).to.be.equal(aPassedIn);
￼            return { b: 4 };
￼          },
￼        };
￼
￼        const rightFullyCovered = {
￼          attachData(context: { b: number }) {
￼            expect(context.b).to.be.equal(bPassedIn);
￼            return { c: 4 };
￼          },
￼        };

￼        const pipedAtoBC = HTPipe(left, rightFullyCovered);
￼
￼        const pipedAtoBCAttachDataReturnValues = await pipedAtoBC.attachData({
￼          a: aToBePassedIn, b: bToBePassedIn,
￼        });
￼
￼        expect(pipedAtoBCAttachDataReturnValues).to.has.property('b');
￼        expect(pipedAtoBCAttachDataReturnValues.b).to.be.equal(4);
￼        expect(pipedAtoBCAttachDataReturnValues).to.has.property('c');
￼        expect(pipedAtoBCAttachDataReturnValues.c).to.be.equal(4);
```

Reply:
done